src/runtime/analytics-service.js

```diff
@@ -130,6 +131,13 @@ export class AnalyticsService {
     // the user wait too long.
     /** @private {?number} */
     this.timeout_ = null;
+
+    // Time source for setting the client timestamp before sending requests.
+    // Injectable for tests.
+    /** @private {function():!../proto/api_messages.Timestamp} */
+    this.timeSource_ = () => {
```

Comment:
Could you call this `getTimestamp_` ? I think that's clearer than `timeSource_`

Reply:
Done.